scripts/utils_clone.groovy

```diff
@@ -0,0 +1,94 @@
+// Clone jobs and view from source
+
+import jenkins.*
+import jenkins.model.*
+import hudson.*
+import hudson.model.*
+import groovy.xml.XmlUtil
+import groovy.lang.*
+import groovy.lang.Binding;
+import groovy.util.XmlSlurper
+import groovy.lang.Tuple2
+import groovy.util.slurpersupport.GPathResult
+import jenkins.model.Jenkins
+import hudson.model.ListView
+
+println '=== Parameters: ==='
+def params = [:]
+build?.actions.find{ it instanceof ParametersAction }?.parameters.each {
+def k = "${it.name}"
+def v = "${it.value}"
+if (v) {
+ params[k] = v
+  println "${k}: ${v}"
+ }
+}
+def user = "luxoft_ci_tech@luxoft.com"
+def token = params["CI_PASSWORD"]
+
+def src
+def trg
+if (params["SOURCE_VIEW"]) { src = params["SOURCE_VIEW"] }
+if (params["TARGET_VIEW"]) { trg = params["TARGET_VIEW"] }
+boolean createIssueJobs = params["CREATE_FEATURE_JOBS"].toBoolean()
+
+if (!src) {
+  println "ERROR: Source view is not defined"
+  return 1
+}
+
+if (!trg) {
+  println "ERROR: Target view is not defined"
+  return 1
+}
+
+Jenkins jenkins = Jenkins.getInstance()
+
+def trgView = new ListView(trg)
+
+jenkins.addView(trgView)
+def reqGet = ["bash", "-c", "curl -X GET http://localhost:8080/view/${src}/config.xml"]
+def process = reqGet.execute()
+process.waitFor()
+def srcCfg = process.text
+def trgCfg = srcCfg.replaceAll(src, trg).replaceAll("[\n\r]", "").trim()
+def reqPost = ["bash", "-c", "echo '${trgCfg}' | curl -X POST -d @- http://localhost:8080/view/${trg}/config.xml -u ${user}:${token}"]
+def process2 = reqPost.execute()
+process2.waitFor()
+
+println "=== New views created: ==="
+println "${trgView.name}"
+
+def srcView = jenkins.getView(src)
+
+println "=== New jobs created: ==="
+for(item in srcView.getItems()) {
+  if (item.name.matches("(.*)_Issue_(.*)") && !createIssueJobs) { continue }
+  def config = item.getConfigFile()
+  File file = config.getFile()
+  String fileContent = file.getText('UTF-8').replaceAll(src, trg)
+  if (item.name.matches("(.*)=RUN=")) {
+    def xml = new XmlParser().parseText(fileContent)
+    def jobParams
+    jobParams.each {
+      it ->
+        def k = "${it.name}"
+        jobParams << k
+    }
```

Comment:
I don't think code in lines `71` - `77` will work.
This part of code originally is required to set job's parameter like `SDL_REPOSITORY`, `SDL_BRANCH` etc. to a new defined values.
However in a current implementation values for all these parameters are not updated. Just source values are copied.
The reason of that is line `72` where `jobParams` variable is created, but not used within `xml` object.

To verify just run `http://172.26.147.12/view/Jenkins_Utils/job/Jenkins_Utils_Clone-clone` with non-default values and check configuration for a new created `*RUN*` jobs.